contracts/AMMStrategy.sol

```diff
@@ -0,0 +1,181 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.17;
+
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { ERC20 } from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import { Ownable } from "./abstracts/Ownable.sol";
+import { Asset } from "./libraries/Asset.sol";
+
+import { IWETH } from "./interfaces/IWeth.sol";
+import { IUniswapPermit2 } from "./interfaces/IUniswapPermit2.sol";
+import { IAMMStrategy } from "./interfaces/IAMMStrategy.sol";
+import { IStrategy } from "./interfaces/IStrategy.sol";
```

Comment:
I think `IStrategy` can be removed here and @inheritdoc using `IAMMStrategy`.

Reply:
There will a compiler error: 
```
Compiler run failed
error[4682]: DocstringParsingError: Documentation tag @inheritdoc references contract "IAMMStrategy", but the contract does not contain a function that is overridden by this function.
  --> contracts/AMMStrategy.sol:97:5:
   |
97 |     /// @inheritdoc IAMMStrategy
```